### ISSUE TITLE
Git: Clone crashes VSCodium

### ISSUE THREAD

**Issue by RavanH** [VSCodium/vscodium#2133]
Type: <b>Bug</b>

VSCodium on Ubuntu, open commands with Ctrl+Shift+P, type "clone" and choose "Git: Clone", then choose "Clone from Github." In the file browser, browse to select another directory to clone the repository. VSCodium crashes  as soon as I try to browse to another directory than the user dir...

Tested without any extensions enabled (except Dutch language extension which I cannot disable for some reason)

Version: 1.95.3
Release: 24321
Commit: dc5a551c256719475d2d0959270e3836c9bbb632
Date: 2024-11-16T01:21:03.577Z
Electron: 32.2.5
ElectronBuildId: undefined
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Linux x64 6.8.0-49-generic snap

VSCodium version: VSCodium 1.95.3 (dc5a551c256719475d2d0959270e3836c9bbb632, 2024-11-16T01:21:03.577Z)
OS version: Linux x64 6.8.0-49-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 PRO 6850U with Radeon Graphics (16 x 1396)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off<br>webnn: unavailable_software|
|Load (avg)|1, 1, 1|
|Memory (System)|11.45GB (6.28GB free)|
|Process Argv|--no-sandbox --force-user-env|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|wayland|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-wordpress-hooks|joh|1.4.0
svn-scm|joh|2.15.2
vscode-language-pack-nl|MS-|1.48.3
php-resolver|Sto|2.1.5
tabnine-vscode|Tab|3.196.0

</details>

**Comment by RavanH** [comment]
I now find that "Add folder to Workspace" or "Open folder" will crash VSCodium immediately. "Open file" does open the file browser, but same as Git Clone, browsing a few folders, VSCodium will crash too...

**Comment by RavanH** [comment]
This appears in the system log after a crash when trying to "Add a folder to Workspace":

codium[61455]: segfault at 7fff1bb08ff8 ip 000073034ea79d36 sp 00007fff1bb09000 error 6 in libgio-2.0.so.0.6400.6[73034e96c000+114000] likely on CPU 15 (core 7, socket 0)

**Comment by RavanH** [comment]
Update: the crashes appear to happen only when browsing to specific folders. Like my ~/Downloads folder or the dedicated ~/projects folder that I use for my projects.

After deleting ALL files from my Dowloads folder, the craches do not happen there anymore. So it appears related to something in the directory that is being opened...

**Comment by daiyam** [comment]
Have you tried with the extensions disabled? or with `codium --verbose`?

**Comment by RavanH** [comment]
> Have you tried with the extensions disabled?
Yes (like I wrote: disabled all except Dutch translation extension) ... to make sure, I uninstalled the Dutch translation extension.

`codium --verbose` show this right before the crash:

```
[main 2024-11-29T00:18:41.995Z] menubar#runActionInRenderer { type: 'commandId', commandId: 'workbench.action.files.openFolder' }
[108083:1129/011842.000241:INFO:CONSOLE(35)] "%cTRACE color: #888 CommandService#executeCommand workbench.action.files.openFolder", source: vscode-file://vscode-app/snap/codium/428/usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js (35)
[main 2024-11-29T00:18:42.002Z] [DialogMainService]: request to acquire file dialog lock {
  title: 'Open Folder',
  buttonLabel: undefined,
  filters: undefined,
  defaultPath: '/home/ravanhagen/Downloads',
  properties: [ 'multiSelections', 'openDirectory', 'createDirectory' ]
}
[main 2024-11-29T00:18:42.003Z] [DialogMainService]: new file dialog lock created {
  title: 'Open Folder',
  buttonLabel: undefined,
  filters: undefined,
  defaultPath: '/home/ravanhagen/Downloads',
  properties: [ 'multiSelections', 'openDirectory', 'createDirectory' ]
}
```

**Comment by RavanH** [comment]
So after some testing, I found that this only happens when there is a json file in the folder. To reproduce:

Create a folder ~/Test and a new file in that folder test.json, with minimal json content like an empty object: `[{}]`

Then run `codium --verbose`, choose Open folder and browse to the Test folder.

The crash with above output occurs.

**Comment by daiyam** [comment]
This is weird! Have you tried with the `.deb`?

Thx for investigating!

**Comment by RavanH** [comment]
@daiyam I just downloaded and installed the deb:

Version: 1.95.3
Release: 24321
Commit: dc5a551c256719475d2d0959270e3836c9bbb632
Date: 2024-11-16T01:21:03.577Z
Electron: 32.2.5
ElectronBuildId: undefined
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Linux x64 6.8.0-49-generic

Lo en behold: no crashes!

Very happy to switch to the DEB version permanently, only: it is not in the Ubuntu repository. Is there a deb repository so I can get updates via the package manager?

**Comment by daiyam** [comment]
> Is there a deb repository so I can get updates via the package manager?

There is https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo

But I plan to use https://build.opensuse.org/ due to some limitation with gitlab (only amd64 and arm64).

**Comment by RavanH** [comment]
Excellent! Thank you :)

Shall I close this, as it apparently only concerns the snap package?

**Comment by RavanH** [comment]
@botlane-feeder 

> My version came from snap, probably as the same version as RavanH.

Yes :)

> And I do have the same problem when I try to "Open Folder", by clicking in the menu or by using the shortcut Ctrl+O.

Is there a JSON file in the folder that is being opened in the browser window?

**Comment by viniciusmelocodes** [comment]
Hello,
I have the same issue, with the version below:
Version: 1.96.0
Release: 24347
Commit: e4195c69747f84abd56a7d1d82fd9fa236511d26
Date: 2024-12-12T16:24:27.456Z
Electron: 32.2.7
ElectronBuildId: undefined
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Linux x64 6.8.0-50-generic snap

I use snap with Ubuntu.

**Comment by RavanH** [comment]
As daiyam suggested, the DEB version from https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo is good. The snap package is the same version but apparently there is some conflict with the snap implementation...

**Comment by schumaml** [comment]
We encountered the same issue in our GIMP AppImage (which happens to be based on Debian Stable right now), and figured out that this is the following bug in GIO/GLib:
https://gitlab.gnome.org/GNOME/glib/-/issues/3191

Fixed in GLIb 2.79.0.

**Comment by lucvtzel** [comment]
Same issue when the file `eslint.config.mjs` is present in the directory. If I delete this file, Codium no longer crashes on `CTRL+O`

Observed with two different snaps:

    Version: 1.95.3
    Release: 24321
    Commit: dc5a551c256719475d2d0959270e3836c9bbb632
    Date: 2024-11-16T01:21:03.577Z
    Electron: 32.2.5
    ElectronBuildId: undefined
    Chromium: 128.0.6613.186
    Node.js: 20.18.0
    V8: 12.8.374.38-electron.0
    OS: Linux x64 6.8.0-57-generic snap

and

    Version: 1.99.22418
    Commit: 58d416f2e6ca39a861dd2d16070d8ceef49764de
    Date: 2025-04-10T18:21:51.641Z
    Electron: 34.3.2
    ElectronBuildId: undefined
    Chromium: 132.0.6834.210
    Node.js: 20.18.3
    V8: 13.2.152.41-electron.0
    OS: Linux x64 6.8.0-57-generic snap

If this can help ... :grey_question:

**Comment by SarenT** [comment]
are there any solutions to this? it seems to be an old problem but impossible to overcome.

**Comment by RavanH** [comment]
> are there any solutions to this? it seems to be an old problem but impossible to overcome.

If you're on Linux, try replacing it with the DEB version from https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo
